$NetBSD: patch-aa,v 1.9 2010/11/14 20:53:03 tonio Exp $

Fix mixed implicit and normal rules
This fixes build with GNU Make 3.82. See threads:
  https://sympa-roc.inria.fr/wws/arc/coqdev/2010-10/msg00025.html
  http://thread.gmane.org/gmane.comp.gnu.make.bugs/4912

Patch from https://gforge.inria.fr/scm/viewvc.php?diff_format=s&view=rev&root=coq&sortby=file&revision=13566

--- Makefile.orig	2010-10-13 19:53:28.000000000 +0000
+++ Makefile
@@ -160,9 +160,19 @@ else
 stage1 $(STAGE1_TARGETS) : always
 	$(call stage-template,1)
 
+ifneq (,$(STAGE1_IMPLICITS))
+$(STAGE1_IMPLICITS) : always
+	$(call stage-template,1)
+endif
+
 stage2 $(STAGE2_TARGETS) : stage1
 	$(call stage-template,2)
 
+ifneq (,$(STAGE2_IMPLICITS))
+$(STAGE2_IMPLICITS) : stage1
+	$(call stage-template,2)
+endif
+
 # Nota:
 # - world is one of the targets in $(STAGE2_TARGETS), hence launching
 # "make" or "make world" leads to recursion into stage1 then stage2
